g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java

```diff
@@ -33,4 +37,97 @@ public static boolean checkParams(String[] categories, String... args){
 
         return true;
     }
+    
+    
+    /**
+     * Get the the matched version from the version list by comparing to the input version
+     *
+     * @param productName
+     * @param version
+     * @return a matched version, if there's no matched version then return input version
+     */
+    public static String getMatchedVersion(final String productName, final String version,  final Map<String, String[]> productsAndVersions) throws L3APIException{
+        String mv = "";
+        if(productsAndVersions != null) {
+            String[] vList = productsAndVersions.get(productName);
+            if(vList != null) {
+                if(Arrays.asList(vList).contains(version)) {
+                    return version;
+                }
+                for(String v : vList) {
+                    if(compareVersion(v, version) == -1 && compareVersion(v, mv) == 1) {
+                        mv = v;
+                    }
+                }
+            }
+        }
+        return StringUtils.isEmpty(mv)? version : mv;
+    }
+    
+    
+    /**
+     * Compare source version and target version
```

Comment:
Is this copied? I think it's not a good pratice to have same code in different places.

Reply:
Because Colin's method is private, I can't use it. If I want to use it, I need to change his codes

Reply:
Could we refactor the code to make public or move to common class? So we can avoid duplication.

Reply:
done